### ISSUE TITLE
CALL METHOD Usage does not check if the method call is dynamic.

### ISSUE THREAD

**Issue by nomssi** [SAP/code-pal-for-abap#396]
**Check Name**
CALL METHOD Usage Check

**Actual Behavior**
Warning "CALL METHOD" statement should not be used! is generated for the following code:

                        CALL METHOD (lo_proc->value)
                          EXPORTING
                            list   = proc_parameters
                          RECEIVING
                            result = result.

**Expected Behavior**
No message should be generated here as there is no alternative syntax for a "dynamic" method call.

**Comment by lucasborin** [comment]
What is the value assigned to the `lo_proc->value`?

**Comment by lucasborin** [comment]
@nomssi: Would you mind testing the solution provided in the branch named `call-method-dynamic`?

**Comment by nomssi** [comment]
I pulled the call-method-dynamic branch.
Now no CALL METHOD messages are generated in my test case (ABAP Scheme, call of a method of the same class.), i.e. the behavior is as expected. I also copied a standard demo for dynamic calls 
    CALL METHOD (class)=>(meth)
      PARAMETER-TABLE
        ptab
      EXCEPTION-TABLE
        etab.
and no CALL METHOD messages were generated.

TLDR:  so far all tests pass.
